src/evict/evict_lru.c

```diff
@@ -354,6 +354,40 @@ __evict_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
     return (ret);
 }
 
+/*
+ * __clear_saved_walk_tree --
+ *     Clear saved walk tree maintaining use count.
+ */
+static void
+__clear_saved_walk_tree(WT_SESSION_IMPL *session)
+{
+    WT_DATA_HANDLE *dhandle;
+
+    dhandle = S2C(session)->cache->walk_tree;
+    if (dhandle == NULL)
+        return;
+    WT_ASSERT(session, __wt_atomic_loadi32(&dhandle->session_inuse) > 0);
+    S2C(session)->cache->walk_tree = NULL;
+    (void)__wt_atomic_subi32(&dhandle->session_inuse, 1);
+}
+
+/*
+ * __set_saved_walk_tree --
+ *     Set saved walk tree maintaining use count.
+ */
+static void
+__set_saved_walk_tree(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
```

Comment:
Is this and the `__clear` function expected to be called under the dhandle list lock?

Reply:
Yeah, it's a great idea to put some asserts around it.

Reply:
No because the dhandle **list** lock guards the dhandle list, not individual dhandles. There's no dhandle lock.